applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -63,30 +67,120 @@ static const struct state_switch state_switch[] = {
 
 #if CONFIG_DESKTOP_BLE_PEER_ERASE
 	{STATE_IDLE,        CLICK_LONG,   STATE_ERASE_PEER,  erase_start},
-	{STATE_ERASE_PEER,  CLICK_DOUBLE, STATE_IDLE,        erase_confirm},
+	{STATE_ERASE_PEER,  CLICK_DOUBLE, STATE_ERASE_ADV,   erase_adv_confirm},
 #endif
 };
 
 
 static enum state state;
 static u32_t timestamp;
-static u8_t cur_peer_id = BT_ID_DEFAULT; /* We expect zero */
-#define PEER_ID_STORAGE_NAME "peer_id"
+static u8_t cur_peer_id;
 
 static u8_t tmp_peer_id;
 static bool button_pressed_on_init;
 
 
+#ifdef CONFIG_BT_PERIPHERAL
+#define BT_STACK_ID_LUT_SIZE CONFIG_BT_MAX_PAIRED
+#elif CONFIG_BT_CENTRAL
+#define BT_STACK_ID_LUT_SIZE 0
+#else
+#error Device must be Bluetooth peripheral or central.
+#endif
+
+static u8_t bt_stack_id_lut[BT_STACK_ID_LUT_SIZE];
+#define TEMP_PEER_ID (CONFIG_BT_MAX_PAIRED - 1)
+
+
 static struct k_delayed_work timeout;
 static struct k_delayed_work single_click;
 static struct k_delayed_work long_click;
 
 
+static u8_t get_bt_stack_peer_id(u8_t id)
+{
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+		if ((state == STATE_ERASE_PEER) ||
+		    (state == STATE_ERASE_ADV)) {
+			return bt_stack_id_lut[TEMP_PEER_ID];
+		} else {
+			return bt_stack_id_lut[id];
+		}
+	} else {
+		/* For central device there is only default identity. */
+		return BT_ID_DEFAULT;
+	}
+}
+
+static int update_bt_stack_id_lut(void)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		char key[MAX_KEY_LEN];
+		int err = snprintk(key, sizeof(key), MODULE_NAME "/%s%d",
+				   BT_ID_LUT_STORAGE_NAME, cur_peer_id);
+
+		if ((err > 0) && (err < MAX_KEY_LEN)) {
+			err = settings_save_one(key, &bt_stack_id_lut[cur_peer_id],
```

Comment:
I believe that entire lut must be saved in one step. If entries are saved separately table may become corrupted in case of power failure between these two writes.

Reply:
you are right, I will change this one